core-bundle/public/ajax-form.js

```diff
@@ -0,0 +1,69 @@
+window.addEventListener('DOMContentLoaded', () => {
+    document.querySelectorAll('[data-ajax-form]').forEach(form => {
+        let triggeredButton = null;
+
+        function request(method, uri, body, callback) {
+            const xhr = new XMLHttpRequest();
+
+            xhr.open(method, uri, true);
+            xhr.setRequestHeader('Accept', 'text/html');
+            xhr.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+            xhr.setRequestHeader('X-Contao-Ajax-Form', form.querySelector('[name="FORM_SUBMIT"]').value);
+
+            form.ariaBusy = 'true';
+            form.dataset.ajaxForm = 'loading';
+
+            xhr.onload = () => {
+                callback(xhr);
+
+                form.ariaBusy = 'false';
+                form.dataset.ajaxForm = '';
+
+                const event = new CustomEvent('ajax-form-onload', {
+                    detail: { body, form, xhr },
+                });
+
+                form.dispatchEvent(event);
+                window.dispatchEvent(event);
```

Comment:
```suggestion
                    detail: { body, form, xhr },
                    bubbles: true,
                });

                form.dispatchEvent(event);
```

If we let the event bubble up the DOM tree we wouldn’t need to dispatch it twice I think.